### DISCUSSION TITLE
How to fix this error?

### DISCUSSION THREAD

**Discussion by Thelistener666** [junjunlab/scRNAtoolVis#18] [General]
运行devtools::install_github('junjunlab/scRNAtoolVis')后出现如下选择
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?
1: All                              
2: CRAN packages only               
3: None                             
4: cli       (3.6.0 -> 3.6.1) [CRAN]
5: htmltools (0.5.4 -> 0.5.5) [CRAN]
选1后出现如下选择
There are binary versions available but the source versions are later:
          binary source needs_compilation
cli        3.6.0  3.6.1              TRUE
htmltools  0.5.4  0.5.5              TRUE
Do you want to install from sources the packages which need compilation? (Yes/no/cancel) 
输入yes

最后报错
ERROR: dependencies ‘ComplexHeatmap’, ‘jjAnno’ are not available for package ‘scRNAtoolVis’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/scRNAtoolVis’
Warning messages:
1: packages ‘jjAnno’, ‘ComplexHeatmap’ are not available for this version of R

Versions of these packages for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages 
2: In i.p(...) : installation of package ‘cli’ had non-zero exit status
3: In i.p(...) :
  installation of package ‘htmltools’ had non-zero exit status
4: In i.p(...) :
  installation of package ‘/var/folders/z_/npq19gzd6dg38_tzfd7lmy800000gn/T//RtmpKef6xF/file128bf9bf87b9/scRNAtoolVis_0.0.5.tar.gz’ had non-zero exit status

**Comment by junjunlab** [comment]
Please choose "3" option to install and install **jjAnno** and **ComplexHeatmap** packages before install scRNAtoolVis.

**Reply by Thelistener666** [reply]
Thanks, but when I install **ComplexHeatmap** by Biocmanager::install("ComplexHeatmap"), it still has some error.

The downloaded binary packages are in
	/var/folders/z_/npq19gzd6dg38_tzfd7lmy800000gn/T//RtmpKef6xF/downloaded_packages
Old packages: 'cli', 'htmltools'
Update all/some/none? [a/s/n]: 
a

  There are binary versions available but the source versions are later:
          binary source needs_compilation
cli        3.6.0  3.6.1              TRUE
htmltools  0.5.4  0.5.5              TRUE

Do you want to install from sources the packages which need compilation? (Yes/no/cancel) Yes
installing the source packages ‘cli’, ‘htmltools’

trying URL 'https://cran.rstudio.com/src/contrib/cli_3.6.1.tar.gz'
Content type 'application/x-gzip' length 567293 bytes (553 KB)
==================================================
downloaded 553 KB

trying URL 'https://cran.rstudio.com/src/contrib/htmltools_0.5.5.tar.gz'
Content type 'application/x-gzip' length 131880 bytes (128 KB)
==================================================
downloaded 128 KB

* installing *source* package ‘cli’ ...
** package ‘cli’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
**ERROR**: compilation failed for package ‘cli’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/cli’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/cli’
* installing *source* package ‘htmltools’ ...
** package ‘htmltools’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
**ERROR**: compilation failed for package ‘htmltools’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/htmltools’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/htmltools’

The downloaded source packages are in
	‘/private/var/folders/z_/npq19gzd6dg38_tzfd7lmy800000gn/T/RtmpKef6xF/downloaded_packages’
Warning messages:
1: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘cli’ had non-zero exit status
2: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘htmltools’ had non-zero exit status

**Comment by junjunlab** [comment]
Try install cli and htmltools locally.
